src/MacroTools/ObjectiveSystem/Objectives/UnitBased/ObjectiveAnyEnemyUnitInRects.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.QuestSystem;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace MacroTools.ObjectiveSystem.Objectives.UnitBased
+{
+  /// <summary>
+  /// Completed when an eligible player moves a unit into the specified <see cref="Rectangle"/>.
+  /// </summary>
+  public sealed class ObjectiveAnyEnemyUnitInRects : Objective, IHasCompletingUnit
+  {
+    private readonly bool _heroOnly;
+    private readonly IEnumerable<Rectangle> _targetRects;
+    private readonly Faction? _faction;
```

Comment:
We don't need this property because we can just use the existing EligibleFactions property instead. Admittedly EligibleFactions is a List which isn't very useful here, but I think you can just apply your check to the first EligibleFaction in the List.